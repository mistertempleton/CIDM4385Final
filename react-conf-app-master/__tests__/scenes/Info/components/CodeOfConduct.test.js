// @flow
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CodeOfConduct
  from '../../../../app/scenes/Info/components/CodeOfConduct';

describe('Info - Code of Conduct', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CodeOfConduct onClose={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
