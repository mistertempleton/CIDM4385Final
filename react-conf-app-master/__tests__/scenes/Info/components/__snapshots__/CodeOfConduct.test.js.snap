exports[`Info - Code of Conduct renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 1334,
        "justifyContent": "flex-end",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }>
    <BlurView
      blurAmount={12}
      blurType="dark"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": 1334,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 750,
          },
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined} />
    </BlurView>
    <View
      style={
        Object {
          "margin": 30,
          "transform": Array [
            Object {
              "scale": 0.93,
            },
            Object {
              "translateY": 100,
            },
          ],
        }
      }>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 6,
            "maxHeight": 400,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 5,
          }
        }>
        <RCTScrollView
          contentContainerStyle={
            Object {
              "padding": 17,
            }
          }>
          <View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4D4D4D",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontSize": 17,
                    },
                  ]
                }>
                Code of Conduct
              </Text>
            </View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                All delegates, speakers and volunteers at React Conf are required to agree with the following code of conduct. Organizers will enforce this code throughout the event.
              </Text>
            </View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4D4D4D",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontSize": 14,
                      "marginTop": 24,
                    },
                  ]
                }>
                The Quick Version
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                Facebook is dedicated to providing a harassment-free conference experience for everyone, regardless of gender, sexual orientation, disability, physical appearance, body size, race, or religion. We do not tolerate harassment of conference participants in any form. Sexual language and imagery is not appropriate for any conference venue, including talks.
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                Conference participants violating these rules may be sanctioned or expelled from the conference without a refund at the discretion of the conference organizers.
              </Text>
            </View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#4D4D4D",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontSize": 14,
                      "marginTop": 24,
                    },
                  ]
                }>
                The Less Quick Version
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                Harassment includes offensive verbal comments related to gender, sexual orientation, disability, physical appearance, body size, race, religion, sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, sustained disruption of talks or other events, inappropriate physical contact, and unwelcome sexual attention.
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                Participants asked to stop any harassing behavior are expected to comply immediately.
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                If a participant engages in harassing behavior, the conference organizers may take any action they deem appropriate, including warning the offender or expulsion from the conference with no refund.
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                If you are being harassed, notice that someone else is being harassed, or have any other concerns, please contact a member of conference staff immediately.
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                Conference staff will be happy to help participants contact venue security or local law enforcement, provide escorts, or otherwise assist those experiencing harassment to feel safe for the duration of the conference. We value your attendance.
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 13,
                    "lineHeight": 17,
                    "marginTop": 14,
                  }
                }>
                We expect participants to follow these rules at all conference venues and conference-related social events.
              </Text>
            </View>
          </View>
        </RCTScrollView>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F2F2F2",
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": -1,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 0,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#999",
                  "fontSize": 24,
                },
                Object {
                  "height": 16,
                  "marginTop": -6,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
